tests/test_history.py

```diff
@@ -308,3 +315,170 @@ def test_construct_history_class(xonsh_builtins):
 def test_construct_history_instance(xonsh_builtins):
     xonsh_builtins.__xonsh__.env["XONSH_HISTORY_BACKEND"] = DummyHistory()
     assert isinstance(construct_history(), DummyHistory)
+
+
+import time
+import calendar
+
+
+HF_FIRST_DAY = calendar.timegm(time.struct_time((2018, 5, 13, 0, 0, 0, 0, 0, 0)))
+
+
+def history_files_list(gen_count) -> (float, int, str, int):
+    """Generate a list of history file tuples"""
+    # generate test list:
+    # 2 files every day in range
+    # morning file has 100 commands, evening 50
+    # first file size 10000, 2nd 2500
+    # first file time 0900, 2nd 2300
+    # for sanity in reproducable test results, all date arithmetic ignores astronomy.
+    # time zone is UTC, all days have 24 h and 0 sec, no leap year or leap sec.
+
+    retval = []
+    for i in range(int((gen_count + 1) / 2)):
+        retval.append(
+            (
+                # first day in sec + #days * 24hr + #hr * 60min + # sec * 60sec + sec= sec to date.
+                HF_FIRST_DAY + (((((i * 24) + 9) * 60) + 0) * 60) + 0,  # mod dt,
+                100,
+                f"/home/bob/.argle/xonsh-{2*i:05n}.json",
```

Comment:
Could this be the reason for Windows failures?

Reply:
Don't think so, these tests don't try to open the history files, just manipulate the list.